bundle/jsky.app.ot/src/main/java/jsky/app/ot/viewer/SPViewer.java

```diff
@@ -893,7 +893,9 @@ public void closeProgram(ISPProgram node) {
     /** Close the current viewer window, but don't exit the application, even if it is the last window. */
     private void closeViewer() {
         if (getRoot() != null) {
-            BagsManager.unwatch(getRoot());
+            for (RootEntry e : _history.rootEntriesAsJava()) {
```

Comment:
Just a note that this could instead be:
`_history.rootEntriesAsJava().stream().foreach(e -> BagsManager.unwatch(e.root()));`